src/main/java/side/util/TaskList.java

```diff
@@ -0,0 +1,188 @@
+package side.util;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import side.tasks.Deadline;
+import side.tasks.Event;
+import side.tasks.Task;
+
+/**
+ * Provides a wrapper around an ArrayList of Tasks to encapsulate the list of tasks given to
+ * Side by its user. Supports operations on tasks in it.
+ *
+ * @author Lua Yi Da
+ */
+
+public class TaskList {
+
+    private ArrayList<Task> tasks;
+    private int taskLabel;
+    private Storage storage;
+
+    /**
+     * Initialises a new TaskList instance with storage and Task arraylist.
+     */
+    public TaskList() {
+        this.tasks = new ArrayList<>();
+        this.taskLabel = 0;
+        this.storage = new Storage();
+    }
+
+    /**
+     * Helper method to retrieve saved tasks from previous session.
+     *
+     * @return String of retrieved tasks.
+     */
+    public String retrieve() {
+        try {
+            String filepath = "./data";
+            this.tasks = this.storage.retrieve(filepath);
+            this.taskLabel = this.tasks.size();
+            return "Here's your history...\n" + this.listToString();
+        } catch (FileNotFoundException e) {
+            return "You have no history...";
+        }
+    }
+
+    /**
+     * Helper method to save tasks for future use. Prints message if IOException occurs.
+     */
+    public void save() {
+        try {
+            String filepath = "./data";
```

Comment:
This path is seen in many places, maybe it would be a good idea to keep it as a higher level constant so as to prevent error in spelling and make it easier to change the path in the future?